sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/DataChangeRecordAction.java

```diff
@@ -91,6 +96,12 @@ public Optional<ProcessContinuation> run(
     outputReceiver.outputWithTimestamp(record, commitInstant);
     watermarkEstimator.setWatermark(commitInstant);
 
+    // The size of a record is represented by the number of bytes needed for the
+    // string representation of the record. Here, we only try to achieve an estimate
+    // instead of an accurate throughput.
+    throughputEstimator.update(
+        Timestamp.now(), record.toString().getBytes(StandardCharsets.UTF_8).length);
```

Comment:
unrelated but this looks unnecessarily expensive. Can you just add a size or bytes method to DataChangeRecord?  The toString appears to include a lot of field names

instead of converting to just get the length, you could use guava method https://guava.dev/releases/19.0/api/docs/com/google/common/base/Utf8.html#encodedLength(java.lang.CharSequence)

Reply:
That is true, let me see if I can modify to get the serialized bytes of the class instead.

Reply:
Since the class is serialized with Avro, I think we could get the serialized form and count the bytes here. We would just need to see how expensive this is compared to the current approach.

I am leaning on leaving this change out of this PR, though, because I am worried we might introduce a bug. Would you be ok if we take this as a fast follow from this change?

Reply:
any concerns with using Utf8.encodedLength for now then?

Reply:
No concerns, made the change.